chained/dialer.go

```diff
@@ -176,10 +173,10 @@ func (p *proxy) MarkFailure() {
 	log.Tracef("Dialer %s consecutive failures: %d -> %d", p.Label(), newCF-1, newCF)
 }
 
-// defaultDialOrigin implements the method from serverConn. With standard proxies, this
+// dialOrigin implements the method from serverConn. With standard proxies, this
 // involves sending either a CONNECT request or a GET request to initiate a
 // persistent connection to the upstream proxy.
-func defaultDialOrigin(op *ops.Op, ctx context.Context, p *proxy, network, addr string) (net.Conn, error) {
+func dialOrigin(op *ops.Op, ctx context.Context, p *proxy, network, addr string) (net.Conn, error) {
```

Comment:
This was always used everywhere, so just use it instead of having a field in proxy